mrmustard/lab/detectors.py

```diff
@@ -246,15 +253,21 @@ class Homodyne(DisplacedSqueezed):
     def __init__(
         self,
         quadrature_angle: Union[float, List[float]],
-        result: Union[float, List[float]] = 0.0,
-        modes: List[int] = None,
+        result: Optional[Union[float, List[float]]],
+        modes: Optional[List[int]] = None,
         r: Union[float, List[float]] = settings.HOMODYNE_SQUEEZING,
     ):
         quadrature_angle = math.astensor(quadrature_angle, dtype="float64")
-        result = math.astensor(result, dtype="float64")
-        x = result * math.cos(quadrature_angle)
-        y = result * math.sin(quadrature_angle)
-        r = math.astensor(r, dtype="float64")
+
+        if result is None:
+            self.sample = True
+            x = 0
+            y = 0
```

Comment:
This values are used to initialize the `DisplacedSqueezed` state and cannot be set to `None` because that class doesn't allow it. They are updated accordingly later during the computation.